biothings/hub/databuild/backend.py

```diff
@@ -154,14 +154,24 @@ def get_src_metadata(self):
             # are sub-source names, not main source so we have to deal with src_dump as well
             # in order to resolve/map main/sub source name
             subsrc_versions = []
+
+            if src and src.get("download"):
+                # Store the latest dump time
+                src_meta.setdefault(src["_id"], {}).setdefault("dump_date", src["download"]["started_at"])
+
             if src and src.get("upload"):
+                latest_upload_date = None
                 meta = {}
                 for job_name in src["upload"].get("jobs", {}):
                     job = src["upload"]["jobs"][job_name]
                     # "step" is the actual sub-source name
                     docm = self.master.find_one({"_id": job.get("step")})
                     if docm and docm.get("src_meta"):
                         meta[job.get("step")] = docm["src_meta"]
+                    # Store the latest upload time
+                    if not latest_upload_date or latest_upload_date < job["started_at"]:
+                        latest_upload_date = job["started_at"]
+                        meta[job.get("step")]["upload_date"] = latest_upload_date
```

Comment:
@sengineer0 similar to "download", what if upload status is not `success`?

Reply:
`@sengineer0 similar to "download", what if upload status is not success?`
The answer is no.
We only store the latest upload information. the old upload is deleted, and replaced with the new one.